pkg/executables/kubectl.go

```diff
@@ -1498,6 +1498,28 @@ func (k *Kubectl) ValidateNodesVersion(ctx context.Context, kubeconfig string, k
 	return nil
 }
 
+func (k *Kubectl) ValidateMachineAndNodeNamesMatch(ctx context.Context, kubeconfig string, namespace string) error {
```

Comment:
we should get some unit tests on here, and check the linting

Reply:
refactored code in test/framework